elastic/kibana
# [ES|QL] Add autocomplete for enrich command

## Summary

Part of #163248

This PR adds the autocomplete feature for the `ENRICH` command:

![esql_enrich_autocomplete](https://github.com/elastic/kibana/assets/924948/80dca2a4-a9cf-4e84-8e5c-e948febffc01)

If there's no available policy saved then a helpful message is shown:

<img width="742" alt="Screenshot 2023-08-25 at 17 23 07" src="https://github.com/elastic/kibana/assets/924948/5b24aa26-81a4-4f1b-a257-d78c548f24a6">

Policy suggestion:

<img width="744" alt="Screenshot 2023-08-25 at 17 55 10" src="https://github.com/elastic/kibana/assets/924948/c1a6a72f-b261-4234-97a3-a4a7af9c4824">
<img width="1004" alt="Screenshot 2023-08-25 at 17 55 04" src="https://github.com/elastic/kibana/assets/924948/6d82c666-78cd-445f-bcc7-76a9fdaf7523">

Matching field suggestion:

<img width="1091" alt="Screenshot 2023-08-25 at 17 55 59" src="https://github.com/elastic/kibana/assets/924948/2b4bbfcd-6e9f-43f3-a8cc-ddae80d20b89">
<img width="828" alt="Screenshot 2023-08-25 at 17 55 54" src="https://github.com/elastic/kibana/assets/924948/06745377-8f2e-480f-94cb-5a612974da7e">

Some UX notes:
* Policy suggestions will show the list of indices defined as info ✅  (`Policy defined on index: ...`)
* after the `ON`  command the list of available fields is shown and the matching field defined in the chosen policy is shown (`Use to match on <matching_field> on the policy`) ✅ 
* after the `WITH` command both a new variable is shown and the list of policy enriched fields ✅ 
  * currently there's no way to detect whether an `enrichIdentifier` is a variable or a `policyEnrichedField` so both an assign and a pipe command is suggested after an identifier ⚠️ 

**This will be merged in the feature branch**

**To test**

Create a policy based on the flights index:

```
PUT /_enrich/policy/my-policy
{
  "match": {
    "indices": "kibana_sample_data_flights",
    "match_field": "OriginCountry",
    "enrich_fields": ["Carrier"]
  }
}

PUT /_enrich/policy/my-policy/_execute
```

Now you're all set to use the `enrich` command in ES|QL. Make sure to use another dataView who has a country code field to match the policy one.

### Checklist

Delete any items that are not applicable to this PR.
